src/managers/TransactionManager/helpers/sessionCallbacks.ts

```diff
@@ -1,18 +1,35 @@
-export const sessionCallbacksMap: SessionCallbacksType = {};
+const sessionCallbacksMap: Record<string, SessionCallbacksType> = {};
 
-type SessionCallbacksType = {
+const genericCallbacksMap: SessionCallbacksType = {};
+
+export type SessionCallbacksType = {
   onSuccess?: (sessionId: string) => Promise<void>;
   onFail?: (sessionId: string) => Promise<void>;
 };
 
-export const registerSessionCallbacks = ({
+export const getCallbacks = (sessionId?: string): SessionCallbacksType => {
+  if (sessionId != null && sessionId in sessionCallbacksMap) {
```

Comment:
Redundant check. if sessionId in sessionCallbacksMap, then for sure sessionId != null